modules/database-commons/src/main/java/org/testcontainers/ext/ScriptScanner.java

```diff
@@ -0,0 +1,151 @@
+ */
+@RequiredArgsConstructor
+class ScriptScanner {
+
+    private final String resource;
+
+    private final String script;
+
+    private final String separator;
+
+    private final String commentPrefix;
+
+    private final String blockCommentStartDelimiter;
+
+    private final String blockCommentEndDelimiter;
+
+    private final Pattern eol = Pattern.compile("[\n\r]+");
+
+    private final Pattern whitespace = Pattern.compile("\\s+");
+
+    private final Pattern identifier = Pattern.compile("[a-z][a-z0-9_]*", Pattern.CASE_INSENSITIVE);
+
+    private final Pattern singleQuotedString = Pattern.compile("'(\\\\'|[^'])*'");
+
+    private final Pattern ansiQuotedString = Pattern.compile("\"(\\\\\"|[^\"])*\"");
+
+    private final Pattern dollarQuotedStringDelimiter = Pattern.compile("\\$\\w*\\$");
+
+    private int offset;
+
+    @Getter
+    private String currentMatch;
+
+    private boolean matches(String substring) {
+        if (script.startsWith(substring, offset)) {
+            currentMatch = substring;
+            offset += currentMatch.length();
+            return true;
+        } else {
+            currentMatch = "";
+            return false;
+        }
+    }
+
+    private boolean matches(Pattern regexp) {
+        Matcher m = regexp.matcher(script);
+        if (m.find(offset) && m.start() == offset) {
+            currentMatch = m.group();
+            offset = m.end();
+            return true;
+        } else {
+            currentMatch = "";
+            return false;
+        }
+    }
+
+    private boolean matchesSingleLineComment() {
+        /* Matches from commentPrefix to the EOL or end of script */
+        if (matches(commentPrefix)) {
+            Matcher m = eol.matcher(script);
+            if (m.find(offset)) {
+                currentMatch = commentPrefix + script.substring(offset, m.end());
+                offset = m.end();
+            } else {
+                currentMatch = commentPrefix + script.substring(offset);
+                offset = script.length();
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private boolean matchesMultilineComment() {
+        /* Matches from blockCommentStartDelimiter to the next blockCommentEndDelimiter.
+         * Error, if blockCommentEndDelimiter is not found. */
+        if (matches(blockCommentStartDelimiter)) {
+            int end = script.indexOf(blockCommentEndDelimiter, offset);
+            if (end < 0) {
+                throw new ScriptUtils.ScriptParseException(
+                    String.format("Missing block comment end delimiter [%s].", blockCommentEndDelimiter),
+                    resource
+                );
+            }
+            end += blockCommentEndDelimiter.length();
+            currentMatch = blockCommentStartDelimiter + script.substring(offset, end);
+            offset = end;
+            return true;
+        }
+        return false;
+    }
+
+    boolean matchesDollarQuotedString() {
```

Comment:
```suggestion
    private boolean matchesDollarQuotedString() {
```

Any reason for package-private?